packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue

```diff
@@ -168,145 +166,118 @@ const codeBlock = {
 const codeBlocks = ref([codeBlock]);
 
 onMounted(async () => {
-	const session = await newSession('beaker', 'Beaker');
-	jupyterSession.value = session;
-
-	session.kernelChanged.connect((_context, kernelInfo) => {
-		const kernel = kernelInfo.newValue;
-		if (kernel?.name === 'beaker') {
-			session.iopubMessage.connect(iopubMessageHandler);
-			setKernelContext();
+	try {
+		const jupyterContext = buildJupyterContext();
+		if (jupyterContext) {
+			await kernelManager.init('beaker', 'Beaker', jupyterContext);
 		}
-	});
+	} catch (error) {
+		console.error('Error initializing Jupyter session:', error);
+	}
 });
 
-onUnmounted(async () => {
+onUnmounted(() => {
 	// Get rid of session/kernels on unmount, otherwise we end upwith lots of kernels hanging
 	// around using up memory.
-	jupyterSession.value?.shutdown();
+	kernelManager.shutdown();
 });
 
-function setKernelContext() {
-	const kernel = jupyterSession.value?.session?.kernel;
-	if (!kernel) {
-		return;
-	}
+function buildJupyterContext() {
 	const contextName =
 		selectedModelFramework.value === ModelFramework.Decapodes ? 'decapodes_creation' : null;
 	const languageName =
 		selectedProgrammingLanguage.value === ProgrammingLanguage.Julia ? 'julia-1.9' : null;
 	if (contextName === null || languageName === null) {
 		console.log("Can't work with current language. Do nothing.");
-		return;
+		return null;
 	}
-	const messageBody = {
-		session: jupyterSession?.value?.name || '',
-		channel: 'shell',
-		content: {
-			context: contextName,
-			language: languageName,
-			context_info: {}
-		},
-		msgType: 'context_setup_request',
-		msgId: createMessageId('context_setup')
+
+	return {
+		context: contextName,
+		language: languageName,
+		context_info: {}
 	};
-	const contextMessage: JupyterMessage = createMessage(messageBody);
+}
 
-	kernel.sendJupyterMessage(contextMessage);
+function setKernelContext() {
+	const jupyterContext = buildJupyterContext();
+	if (jupyterContext) {
+		kernelManager.sendMessage('context_setup_request', jupyterContext);
+	}
 }
 
 function handleCode() {
-	const kernel = jupyterSession.value?.session?.kernel;
-	if (!kernel) {
-		return;
-	}
 	isProcessing.value = true;
 
 	const code = editor.value?.getValue();
-	const messageBody: JSONObject = {
-		session: jupyterSession?.value?.name || 'shell',
-		channel: 'shell',
-		content: <JSONObject>{
-			declaration: code
-		},
-		msgType: 'compile_expr_request',
-		msgId: createMessageId('context_setup')
+	const messageContent = <JSONObject>{
+		declaration: code
 	};
-	const compileExprMessage: JupyterMessage = createMessage(messageBody);
 
 	modelValid.value = false;
-	kernel.sendJupyterMessage(compileExprMessage);
+
+	const compileExprMessage = kernelManager.sendMessage('compile_expr_request', messageContent);
```

Comment:
Are those message define somewhere as an ENUM? If not could we do so?

Reply:
They are not. It remains to be seen whether enum makes sense or not, because so far these are very specific request that will only get called once.  E.g. you will unlikely have another instance of `compile_expr_request` anywhere in Terarium.